tools/cli/internal/changelog/merge.go

```diff
@@ -0,0 +1,323 @@
+package changelog
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/mongodb/openapi/tools/cli/internal/changelog/outputfilter"
+	"github.com/tufin/oasdiff/checker"
+)
+
+const (
+	endpointAddedCode       = "endpoint-added"
+	endpointDeprecatedCode  = "endpoint-deprecated"
+	endpointReactivatedCode = "endpoint-reactivated"
+	notSetPriority          = 10
+	changeTypeRelease       = "release"
+	changeTypeUpdate        = "update"
+	changeTypeDeprecated    = "deprecate"
+	changeTypeRemove        = "remove"
+	ChangeLevelErr          = 3
+	ChangeLevelWarn         = 2
+	ChangeLevelInfo         = 1
+)
+
+func newChangeTypePriority() map[string]int {
+	return map[string]int{
+		changeTypeRemove:     1,
+		changeTypeRelease:    1,
+		changeTypeUpdate:     2,
+		changeTypeDeprecated: 3,
+	}
+}
+
+func newChangeTypeOverrides() map[string]string {
+	return map[string]string{
+		endpointAddedCode: changeTypeRelease,
+	}
+}
+
+func (m *Metadata) MergeChangelog() ([]*Entry, error) {
+	changes, err := m.newOasDiffEntries()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(changes) == 0 {
+		return m.BaseChangelog, nil
+	}
+
+	conf := outputfilter.NewOperationConfigs(m.Base, m.Revision)
+	return m.mergeChangelog(changeTypeUpdate, changes, conf)
+}
+
+func (m *Metadata) mergeChangelog(
```

Comment:
can we explicitly filter the path changes somewhere here, or we don't need it?

Reply:
we filter the path changes inside `newPathsFromChanges`